task/sql-tasks.js

```diff
@@ -281,7 +470,34 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        WITH productInfo AS (
+            SELECT DISTINCT
+                cust.CompanyName,
+                orddt.productID,
+                orddt.UnitPrice,
+                prod.ProductName
+            FROM Customers cust, Orders ord, OrderDetails orddt, Products prod
+            WHERE ord.CustomerID = cust.CustomerID 
+            AND orddt.OrderID = ord.OrderID 
+            AND  prod.productID = orddt.productID
+        ), maxValues AS (
+            SELECT
+                CompanyName,
+                    MAX(UnitPrice)
+            AS maxPrice
+            FROM productInfo
+            GROUP BY CompanyName
+        )
+        SELECT
+            p.CompanyName,
+            p.ProductName,
+            p.UnitPrice AS PricePerItem
+        FROM productInfo p, maxValues max
+        WHERE p.companyname = max.companyname AND  p.unitprice = max.maxPrice

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.